### PR TITLE
Started to implement bar handlers

### PR DESCRIPTION
**Pull request by gpyh**
`bar_cmd_bindsym` is my first victim.
I didn't figure out how to properly send an error when parsing `argv[1]` (i. e. `buttonX`).

And, not that I like dissing myself, but I may suck.

### CODE REVIEW

File: include/config.h [hunk 1]
```diff
@@ -73,6 +81,7 @@ struct bar_config {
 	 */
 	char *hidden_state;
 	uint32_t modifier;
+        list_t *bindings;
```
Comment (SirCmpwn) [inline include/config.h]: Spaces!

File: sway/commands.c [hunk 2]
```diff
@@ -1479,31 +1496,100 @@ static struct cmd_handler handlers[] = {
...
+	/* { "strip_workspace_numbers", bar_cmd_strip_workspace_numbers }, */
 	{ "tray_output", NULL },
+	/* { "tray_output", bar_cmd_tray_output }, */
 	{ "tray_padding", NULL },
+	/* { "tray_padding", bar_cmd_tray_padding }, */
 	{ "workspace_buttons", NULL },
+	/* { "workspace_buttons", bar_cmd_workspace_buttons }, */
 };
 
+
```
Comment (SirCmpwn) [inline sway/commands.c]: I don't like all this commented out code. Just add the code when you implement the thing.

**Comment by SirCmpwn** [comment]
Looks good save for the minor issues I mentioned.

**Comment by SirCmpwn** [comment]
Spaces issue is everywhere, by the way. Use tabs.

**Comment by SirCmpwn** [comment]
Oh, don't return nothing. Implement those error cases!

**Comment by gpyh** [comment]
Styling is fixed.
I didn't return anything because I didn't know how.
I'll look into it tomorrow.

File: sway/config.c [hunk 3]
```diff
@@ -489,3 +489,33 @@ void free_sway_binding(struct sway_binding *binding) {
...
+	if ((cmp = sway_binding_cmp_keys(a, b)) != 0) {
+		return cmp;
+	}
+	const struct sway_mouse_binding *binda = a, *bindb = b;
+	return lenient_strcmp(binda->command, bindb->command);
+}
+
+void free_sway_mouse_binding(struct sway_mouse_binding *binding) {
+        //TODO: AFAIK freeing the button field doesn't make sense
+        //but better be safe than sorry, so ask first
```
Comment (auscompgeek) [inline sway/config.c]: You missed the spaces in this file.

File: sway/commands.c [hunk 4]
```diff
@@ -1526,6 +1527,41 @@ static struct cmd_handler handlers[] = {
...
+		return cmd_results_new(CMD_INVALID, "bindsym", "Invalid mouse binding %s", state);
+	}
+	if (strncmp(argv[1], "button", 6) != 0) {
+		return cmd_results_new(CMD_INVALID, "bindsym", "Invalid mouse binding %s", state);
+	}
+	struct sway_mouse_binding *binding = malloc(sizeof(struct sway_mouse_binding));
+	binding->button = numbutton;
+	binding->command = join_args(argv + 1, argc - 1);
+
+	struct bar_config bar = config->bar;
```
Comment (mikkeloscar) [inline sway/commands.c]: use `config->current_bar` to reference the correct bar.

File: sway/config.c [hunk 5]
```diff
@@ -517,3 +517,33 @@ void free_sway_binding(struct sway_binding *binding) {
...
+	const struct sway_mouse_binding *binda = a, *bindb = b;
+
+		if (binda->button > bindb->button) {
+		  return 1;
+		} 
+		if (binda->button < bindb->button) {
+		  return -1;
+		}
+		return 0;
+}
```
Comment (SirCmpwn) [inline sway/config.c]: Indentation here is a bit off.

File: sway/config.c [hunk 6]
```diff
@@ -517,3 +517,33 @@ void free_sway_binding(struct sway_binding *binding) {
...
+	if ((cmp = sway_binding_cmp_keys(a, b)) != 0) {
+		return cmp;
+	}
+	const struct sway_mouse_binding *binda = a, *bindb = b;
+	return lenient_strcmp(binda->command, bindb->command);
+}
+
+void free_sway_mouse_binding(struct sway_mouse_binding *binding) {
+		//TODO: AFAIK freeing the button field doesn't make sense
+		//but better be safe than sorry, so ask first
```
Comment (SirCmpwn) [inline sway/config.c]: Here as well.

Comment (SirCmpwn) [inline sway/config.c]: This comment doesn't make sense anyway, the button is a statically allocated uint32_t.

File: sway/commands.c [hunk 7]
```diff
@@ -1727,12 +1763,14 @@ static struct cmd_handler bar_handlers[] = {
 	{ "workspace_buttons", bar_cmd_workspace_buttons },
 };
 
+
```
Comment (mikkeloscar) [inline sway/commands.c]: No newline needed here :)

File: sway/commands.c [hunk 8]
```diff
@@ -1526,6 +1527,41 @@ static struct cmd_handler handlers[] = {
...
+		sway_log(L_DEBUG, "bindsym - '%s' for swaybar already exists, overwriting", argv[0]);
+		struct sway_mouse_binding *dup = bar.bindings->items[i];
+		free_sway_mouse_binding(dup);
+		list_del(bar.bindings, i);
+	}
+	list_add(bar.bindings, binding);
+	list_sort(bar.bindings, sway_mouse_binding_cmp);
+
+	sway_log(L_DEBUG, "bindsym - Bound %s to command %s when clicking swaybar", argv[0], binding->command);
+
```
Comment (SirCmpwn) [inline sway/commands.c]: Err, does this function end?

File: sway/commands.c [hunk 9]
```diff
@@ -1727,12 +1763,14 @@ static struct cmd_handler bar_handlers[] = {
...
 };
 
+
 static int handler_compare(const void *_a, const void *_b) {
 	const struct cmd_handler *a = _a;
 	const struct cmd_handler *b = _b;
 	return strcasecmp(a->command, b->command);
 }
 
+
```
Comment (mikkeloscar) [inline sway/commands.c]: And here :)

**Pull request closed without merging**